mscore/events.cpp

```diff
@@ -849,7 +849,14 @@ void ScoreView::contextMenuEvent(QContextMenuEvent* ev)
       QPoint gp          = ev->globalPos();
       editData.startMove = toLogical(ev->pos());
       editData.buttons   = Qt::NoButton;
-      Element* e         = elementNear(editData.startMove);
+      Element* e         = nullptr;
+      if (ev->reason() == QContextMenuEvent::Keyboard) {
+            e = score()->selection().element();
+            if (!e)
+                  e = score()->selection().findMeasure();
```

Comment:
It would probably be better to put handling of this case to ["else" branch](https://github.com/musescore/MuseScore/blob/7278beddfaf6bab151fbcc896e6ed01b185a6dce/mscore/events.cpp#L868-L881) since it would call `measurePopup()` rather than `objectPopup()` in this case, which seems to be more suitable. This is actually what I wrote about in my comment above.

Reply:
I see now.  I was thinking the goal was to only change behavior for the keyboard case, but you're actually talking about changing the behavior of right-click any time there is no nearby element, to try to use the selection.  That makes sense.

It's possible, I suppose, someone might prefer the current behavior where right-clicking an empty area of the score always give you access to the style & page settings dialogs, but I think the proposed behavior does make more sense, and you can always get the old behavior back by pressing Esc first.

I will update, and re-remove the disabled code (snuck back in because I was developing & testing on two different machines).